### ISSUE TITLE
Implement fancier indexing for MPO/MPS

### ISSUE THREAD

**Issue by kshyatt** [ITensor/ITensors.jl#61]
Would be nice to be able to do something like

```
M = MPS(10)
M[1] = a_tensor
M[2:5] = my_tensors
M[6:10] = your_tensors
```

**Comment by mtfishman** [comment]
Hi Katie,

Was there a particular use case you had for getting a subset of ITensors from an MPS? Also, should it output a `vector{ITensor}`, another MPS of that smaller size, or some sort of "view" into that section of the MPS? I think the application would help determine what the output should be.

Cheers,
Matt

**Comment by mtfishman** [comment]
This was implemented in #477.
